modules/unsupported/mbstyle/src/main/java/org/geotools/mbstyle/layer/SymbolMBLayer.java

```diff
@@ -1672,32 +1680,55 @@ public Expression textTranslateAnchor() {
         String symbolPlacementVal =
                 transformer.requireLiteral(
                         symbolPlacement(), String.class, "point", "symbol-placement", getId());
-
+        Expression fontSize = textSize();
         if ("point".equalsIgnoreCase(symbolPlacementVal.trim())) {
             // Point Placement (default)
             PointPlacement pointP = sb.createPointPlacement();
             // Set anchor point (translated by text-translate)
             // GeoTools AnchorPoint doesn't seem to have an effect on PointPlacement
             pointP.setAnchorPoint(anchorPoint());
 
-            // MapBox text-offset: +y means down
-            Displacement textTranslate = textTranslateDisplacement();
-            textTranslate.setDisplacementY(
-                    ff.multiply(ff.literal(-1), textTranslate.getDisplacementY()));
-            pointP.setDisplacement(textTranslate);
-
+            // MapBox text-displacement: +y means down, expressed in px
+            Displacement displacement = null;
+            if (hasTextTranslate()) {
+                Displacement textTranslate = textTranslateDisplacement();
+                textTranslate.setDisplacementY(
+                        ff.multiply(ff.literal(-1), textTranslate.getDisplacementY()));
+                displacement = textTranslate;
+            }
+            // MapBox test-offset: +y mean down and expressed in ems
```

Comment:
Please update our copy of the spec to have the correct units - https://github.com/geotools/geotools/blob/master/docs/user/unsupported/mbstyle/spec.rst#text-offset
(I assume the mapbox style spec changed)

Reply:
Does not look like it did, text-offset unit has been "ems" since 5 years ago, at least according to the blame of the spec json file:
https://github.com/mapbox/mapbox-gl-js/blame/master/src/style-spec/reference/v8.json#L2226

Reply:
Weird; we've got pixel in our copy, and most of that was copied directly from the original mapbox spec and then modified. 

Digging into history, looks like ems is correctly listed here: https://github.com/geotools/geotools/commit/26a04bf3808ec97af1a1ecbecb201c5545108164#diff-ecd0a522d59eabbfd572b4d8aacbea57R459
But in the same commit, our copy of the spec [has pixel](https://github.com/geotools/geotools/commit/26a04bf3808ec97af1a1ecbecb201c5545108164#diff-b78844df4f739937004b933448e36c39R3676), so I guess there was an error in our end.